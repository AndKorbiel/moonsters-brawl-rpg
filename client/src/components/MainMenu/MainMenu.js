import MenuList from '@mui/material/MenuList';
import MenuItem from '@mui/material/MenuItem';
import ListItemText from '@mui/material/ListItemText';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';

export default function MainMenu(props) {
    return (
        <Box maxWidth="sm" className="centered text-centered main-menu" data-testid="main-menu">
            <MenuList>
                {props.options.map(element => {
                    return (
                        <MenuItem key={element.value}>
                            <ListItemText>
                                <Button variant="h2" onClick={()=> props.action(element.value)}>
                                    {element.label}
                                </Button>
                            </ListItemText>
                        </MenuItem>
                    )
                })}
            </MenuList>
        </Box>
    )
}