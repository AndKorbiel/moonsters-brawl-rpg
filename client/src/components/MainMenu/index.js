export * from './About';
export * from './HighScore';
export * from './LoadGame';
export * from '../../modules/MainMenu';
export * from './SaveGame';
