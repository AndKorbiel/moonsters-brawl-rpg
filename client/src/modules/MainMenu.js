import MenuList from '@mui/material/MenuList';
import MenuItem from '@mui/material/MenuItem';
import ListItemText from '@mui/material/ListItemText';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';

export function MainMenu({ options, action }) {
  return (
    <Box
      maxWidth="sm"
      className="centered text-centered main-menu"
      data-testid="main-menu"
    >
      <MenuList>
        {options.map((option) => {
          return (
            <MenuItem key={option.value}>
              <ListItemText>
                <Button variant="h2" onClick={() => action(option.value)}>
                  {option.label}
                </Button>
              </ListItemText>
            </MenuItem>
          );
        })}
      </MenuList>
    </Box>
  );
}
