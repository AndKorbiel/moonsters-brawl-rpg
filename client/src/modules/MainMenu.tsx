import MenuList from '@mui/material/MenuList';
import MenuItem from '@mui/material/MenuItem';
import ListItemText from '@mui/material/ListItemText';
import Button from '@mui/material/Button';

import { useDispatch } from 'react-redux';
import { setStatusCode } from '../redux/actions/game';
import React from 'react';
import { Box } from '@mui/material';
import { MenuOption } from '../types';

export function MainMenu({ options }: { options: MenuOption[] }) {
  const dispatch = useDispatch();

  return (
    <Box
      maxWidth="sm"
      className="centered text-centered main-menu"
      data-testid="main-menu"
    >
      <MenuList>
        {options.map((option) => {
          return (
            <MenuItem key={option.value}>
              <ListItemText>
                <Button
                  variant="text"
                  onClick={() => dispatch(setStatusCode(option.value))}
                >
                  {option.label}
                </Button>
              </ListItemText>
            </MenuItem>
          );
        })}
      </MenuList>
    </Box>
  );
}
