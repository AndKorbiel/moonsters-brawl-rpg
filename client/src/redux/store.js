import { createStore, applyMiddleware, compose } from 'redux';
import { rootReducer } from './reducers/index.js';
import thunk from 'redux-thunk';

export const store = createStore(rootReducer, compose(applyMiddleware(thunk)));
