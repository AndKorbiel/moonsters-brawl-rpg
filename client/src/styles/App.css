.App {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  color: #fff;
}

.navbar {
  background: #b5d552;
  position: fixed;
  width: 100%;
  color: #000;
  padding: 10px 0;
  z-index: 99;
}

.MuiTabs-flexContainer {
  align-items: center;
  justify-content: center;
}


.main-cont {
  padding-top: 80px;
}