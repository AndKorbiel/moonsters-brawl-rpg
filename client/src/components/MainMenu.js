import Paper from '@mui/material/Paper';
import MenuList from '@mui/material/MenuList';
import MenuItem from '@mui/material/MenuItem';
import {connect} from 'react-redux';
import {changeStatus} from "../redux/actions";

function MainMenu(props) {
    const handleChange = (value) => {
        props.changeStatus(value)
    }

    return (
        <Paper sx={{ width: 320, maxWidth: '100%' }}>
            <MenuList>
                <MenuItem onClick={() => handleChange('New Game')}>
                    <h2>New game</h2>
                </MenuItem>
                <MenuItem>
                    <h2>Load game</h2>
                </MenuItem>
            </MenuList>
        </Paper>
    )
}

const mapDispatchToProps = dispatch => {
    return {
        changeStatus: newStatus => dispatch(changeStatus(newStatus))
    }
}

export default connect(null, mapDispatchToProps)(MainMenu)