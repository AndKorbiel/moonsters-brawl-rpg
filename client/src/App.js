import './styles/App.scss';
import Grid from '@mui/material/Grid';
import { ThemeProvider, createTheme } from '@mui/material/styles';

import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { START_GAME } from './redux/constants/game';

import { FightScreenCard, MainMenu, OpponentCard, ShopCard } from './modules/';
import { CharacterCardContainer } from './modules/CharacterCard';
import {
  About,
  HighScore,
  LoadGame,
  Login,
  Navbar,
  SaveGame,
} from './components';

const theme = createTheme({
  palette: {
    primary: {
      main: '#2c0f23',
    },
    secondary: {
      main: '#fe7800',
    },
  },
});

export default function App() {
  const dispatch = useDispatch();
  const game = useSelector((state) => state.game);
  const { menuOptions, statusCode } = game;

  useEffect(() => {
    if (statusCode === 1) dispatch({ type: START_GAME });
  }, [dispatch, statusCode]);

  if (!menuOptions) return <h1>Loading...</h1>;

  return (
    <ThemeProvider theme={theme}>
      <Navbar status={statusCode} />

      <div className="App">
        <Grid
          container
          maxWidth="xl"
          spacing={2}
          className="main-cont centered"
        >
          {statusCode === 0 && (
            <Grid item xs={12}>
              <MainMenu options={menuOptions} />
            </Grid>
          )}

          {statusCode === 1 && (
            <>
              <Grid item xs={12} md={4} lg={3}>
                <CharacterCardContainer />
              </Grid>
              <Grid item xs={12} md={8} lg={9}>
                <ShopCard />
              </Grid>
            </>
          )}

          {statusCode === 2 && (
            <>
              <Grid item xs={12} md={4} lg={3}>
                <CharacterCardContainer />
              </Grid>
              <Grid item xs={12} md={4} lg={6}>
                <FightScreenCard />
              </Grid>
              <Grid item xs={12} md={4} lg={3}>
                <OpponentCard />
              </Grid>
            </>
          )}

          {statusCode === 5 && <SaveGame />}
          {statusCode === 6 && <LoadGame />}
          {statusCode === 7 && <About />}
          {statusCode === 8 && <HighScore />}
          {statusCode === 9 && <Login />}
        </Grid>
      </div>
    </ThemeProvider>
  );
}
