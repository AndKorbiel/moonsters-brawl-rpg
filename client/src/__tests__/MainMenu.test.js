import React from 'react';
import MainMenu from "../components/MainMenu";
import {render, screen} from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom';
import {Provider} from "react-redux";
import {store} from "../redux/store";
import App from "../App";

const menuOptions= [
    {value: 1, label: "New Game"},
    {value: 5, label: "Save Game"},
    {value: 6, label: "Load Game"},
    {value: 7, label: "Quit"}
]

it('renders correctly with props', ()=> {
    render(<MainMenu options={menuOptions} />)
})

