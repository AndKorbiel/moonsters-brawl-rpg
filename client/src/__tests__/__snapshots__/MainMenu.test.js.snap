// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="centered text-centered MuiBox-root css-1274r9c"
  data-testid="main-menu"
>
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    onKeyDown={[Function]}
    role="menu"
    tabIndex={-1}
  >
    <li
      className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <button
            className="MuiButton-root MuiButton-h3 MuiButton-h3Primary MuiButton-sizeMedium MuiButton-h3SizeMedium MuiButtonBase-root  css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            New Game
          </button>
        </span>
      </div>
    </li>
    <li
      className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <button
            className="MuiButton-root MuiButton-h3 MuiButton-h3Primary MuiButton-sizeMedium MuiButton-h3SizeMedium MuiButtonBase-root  css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Save Game
          </button>
        </span>
      </div>
    </li>
    <li
      className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <button
            className="MuiButton-root MuiButton-h3 MuiButton-h3Primary MuiButton-sizeMedium MuiButton-h3SizeMedium MuiButtonBase-root  css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Load Game
          </button>
        </span>
      </div>
    </li>
    <li
      className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <button
            className="MuiButton-root MuiButton-h3 MuiButton-h3Primary MuiButton-sizeMedium MuiButton-h3SizeMedium MuiButtonBase-root  css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Quit
          </button>
        </span>
      </div>
    </li>
  </ul>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="centered text-centered MuiBox-root css-1274r9c"
  data-testid="main-menu"
>
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    onKeyDown={[Function]}
    role="menu"
    tabIndex={-1}
  >
    <li
      className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <button
            className="MuiButton-root MuiButton-h3 MuiButton-h3Primary MuiButton-sizeMedium MuiButton-h3SizeMedium MuiButtonBase-root  css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            New Game
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </li>
    <li
      className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <button
            className="MuiButton-root MuiButton-h3 MuiButton-h3Primary MuiButton-sizeMedium MuiButton-h3SizeMedium MuiButtonBase-root  css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Save Game
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </li>
  </ul>
</div>
`;
